src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -95,6 +99,31 @@ public static boolean shouldBeHearingsInWelshFlag() {
 
     public static List<HearingLocation> getHearingLocations(SscsCaseData caseData,
                                                             ReferenceDataServiceHolder referenceDataServiceHolder) {
+        List<HearingLocation> hearingLocations = getAllHearingLocations(caseData, referenceDataServiceHolder);
+
+        String nextHearingVenueName = caseData.getAdjournCaseNextHearingVenue();
+
+        //TODO: remove flag
+        if (referenceDataServiceHolder.isAdjournmentFlagEnabled() && isNotEmpty(nextHearingVenueName)
+            && !HearingsChannelMapping.isPaperCase(caseData)) {
+            try {
+                String venueID = getVenueID(caseData, nextHearingVenueName);
+
+                log.info("Getting hearing locations with the venue ID of {}", venueID);
+
+                return hearingLocations.stream()
+                    .filter(location -> location.getLocationId().equals(venueID))
+                    .collect(Collectors.toList());
```

Comment:
I think it would be better to do
```suggestion
                return HearingLocation.builder()
                    .locationId(epimsId)
                    .locationType(COURT)
                    .build()
```